ember-server/src/main/scala/org/http4s/ember/server/internal/ServerHelpers.scala

```diff
@@ -151,18 +152,78 @@ private[server] object ServerHelpers {
         }
         .drain
 
-    sg.server[F](bindAddress, additionalSocketOptions = additionalSocketOptions)
+    val handler = sg.server[F](bindAddress, additionalSocketOptions = additionalSocketOptions)
       .interruptWhen(shutdown.signal.attempt)
-      // Divorce the scopes of the server stream and handler streams so the
-      // former can be terminated while handlers complete.
-      .prefetch
       .map { connect =>
         shutdown.trackConnection >>
           Stream
             .resource(connect.flatMap(upgradeSocket(_, tlsInfoOpt)))
             .flatMap(withUpgradedSocket(_))
       }
-      .parJoin(maxConcurrency)
-      .drain
+      
+    forking(handler)
+  }
+
+  // Similar to parJoin with a few semantic differences:
+  // - Inner stream values are not emitted to the new stream
+  // - The outer stream can terminate and finalize early
```

Comment:
This means that any resources bound to the scope of the outer stream should _not_ be used in the inner stream, which makes sense for these network servers, but it's a pretty huge caveat

Reply:
cool, I was about to comment that this is the reason why `parJoin` behaves like it does (and why it needs to mess around with scope leasing to achieve that behaviour), but you got it already. I think it's worth adding it as a code comment too